### DISCUSSION TITLE
The location of cue

### DISCUSSION THREAD

**Discussion by quovadis-7** [cue-lang/cue#1387] [Q&A]
After installing cue, where can I find the location of cue in my computer?

**Comment by verdverm** [comment]
That likely depends on the installation method and operating system. Can you provide more details?

*nix type systems have `which <program>` that will display this based on using the PATH variable

**Comment by quovadis-7** [comment]
> That likely depends on the installation method and operating system. Can you provide more details?
> 
> *nix type systems have `which <program>` that will display this based on using the PATH variable

In mac OS or linux operating system, can you tell me the location in all possible scenarios, including:
binary install; from source(through go). I want to know whether the user has installed cue, so they can use cue fmt in the command line.

**Comment by verdverm** [comment]
> can you tell me the location in all possible scenarios

Generally speaking, no.

> I want to know whether the user has installed cue

The `which` tool can tell you this. Many languages have a similar helper in their stdlib. You can also try running `cue help` and catch what happens.

> so they can use cue fmt in the command line

It will have to be somewhere in their `PATH`. That should make things easier for you
